src/test/ui/unboxed-closures/unboxed-closure-sugar-wrong-trait.stderr

```diff
@@ -1,8 +1,16 @@
-error[E0107]: wrong number of type arguments: expected 0, found 1
+error[E0107]: this trait takes 0 type arguments but 1 type argument was supplied
```

Comment:
Ideally we would give the error for `non-Fn trait using parenthetical notation` higher priority and emit it _instead_ of these two, but that should be done in a different PR.